eeps/eep-0072.md

```diff
@@ -0,0 +1,227 @@
+using Erlang records) should naturally have a `#div` record, however, due to
+'div' being a reserved word (the integer division operator), the record `#panel`
+is used instead to save the programmer from having to invoke `#'div'`,
+which feels unnatural and awkward.
+
+Further, applications such as ASN.1 and Corba both have naming conventions that
+rely heavily on uppercase record names and as such, they currently must be
+quoted as well. You can see this in modules in Erlang's
+[`asn1`](https://github.com/erlang/otp/blob/OTP-27.1.1/lib/asn1/src/asn1_records.hrl#L35-L39)
+application. (The previous link points to some record definitions in `asn1`,
+but you can see the usage scattered across a number of modules in the `asn1`
+application).
+
+Usage Syntax Specification
+==========================
+
+This EEP simplifies the above example by
+
+1. Allowing reserved words and variables to be used without quotes for record
+   names, and
+2. Simplifying the definition such that the syntax between record definition
+   and record usage becomes more consistent.
+
+With the changes from this EEP, the above code becomes:
+
+```erlang
+-record('div', {a, b}).
+
+go() ->
+    X = #div{a = 1, b = 2},
+    Y = X#div{a = something_else},
+    Z = Y#div.a,
+    ...
+```
+
+Definition Syntax Motivation
+============================
+
+While the updated example in the usage syntax specification makes the *using*
+of records cleaner, there remains one more inconsistency that can also be
+relatively easily solved. That is the record definition still also needing to
+quote record name, as the example above demonstrates (repeated here for
+convenience):
+
+```erlang
+-record('div', {a, b}).
+
+go() ->
+    X = #div{a = 1, b = 2},
+    Y = X#div{a = something_else},
+    Z = Y#div.a,
+    ...
+```
+
+So whereas the record definition needs to be thought of as `'div'`, the record
+usage no longer requires the quoted term 'div', which could certainly lead an
+Erlang beginner to wonder why 'div' needs to be quoted in the definition while
+other atom-looking terms don't.
+
+Definition Syntax Specification
+===============================
+
+Conveniently, there is a rather easy solution, and that's to
+allow the record usage syntax to also be used as the record definition.
+
+This EEP also then also adds a new record definition syntax, improving the
+symmetry between general record usage and record definition.
+
+The above example can fully then look like the following:
+
+```erlang
+-record #div{a, b}.
+
+go() ->
+    X = #div{a = 1, b = 2},
+    Y = X#div{a = something_else},
+    Z = Y#div.a,
+    ...
+```
+
+Implementation
+==============
+
+To update the syntax for using records, we can safely augment the parser to
+change `# atom` to `# record_name` and define `record_name` to be either
+`atom`, `var`, or `reserved_word`.
+
+To update the record definition syntax, we can simply add a few new
+modifications to the `attribute` Nonterminal.
```

Comment:
```suggestion
modifications to the `attribute` Nonterminal to allow `'#' record_name` as name for the `record` attribute, instead of `atom` as for generic attributes.
```